src/background/controller/wallet.ts

```diff
@@ -1759,10 +1761,9 @@ export class WalletController extends BaseController {
     await this.getNetwork();
     const amountStr = amount.toString();
 
-    const amountBN = new BN(amountStr.replace('.', ''));
+    let amountBN = new BN(amountStr);
 
-    const decimalsCount = amountStr.split('.')[1]?.length || 0;
-    const scaleFactor = new BN(10).pow(tokenResult!.decimals - decimalsCount);
+    const scaleFactor = new BN(10).pow(new BN(tokenResult.decimals));
 
```

Comment:
Could add some safety checks to be sure decimals is < 77

const decimals = tokenResult.decimals ?? 18; // fallback to 18 if undefined
if (decimals < 0 || decimals > 77) { // 77 is BN.js max safe decimals
    throw new Error('Invalid decimals');
}